packages/lesswrong/lib/collections/posts/views.ts

```diff
@@ -840,26 +839,27 @@ ensureIndex(Posts,
 );
 
 Posts.addView("nearbyEvents", (terms: PostsViewTerms) => {
-  const yesterday = moment().subtract(1, 'days').toDate();
   const onlineEvent = terms.onlineEvent === false ? false : viewFieldAllowAny
   let query: any = {
     selector: {
       location: {$exists: true},
       groupId: null,
       isEvent: true,
       onlineEvent: onlineEvent,
-      $or: [{startTime: {$exists: false}}, {startTime: {$gt: yesterday}}],
+      $or: [{startTime: {$gt: new Date()}}, {endTime: {$gt: new Date()}}],
```

Comment:
@jpaddison3 I realized that I needed to exclude events without a `startTime`, because the sort is by `startTime` asc and that put all the events **without** a time at the top of the list. As far as I can tell, there is no way to tell mongo to put null values at the end of the list, and I don't really think it's worth doing a second query for them. This does mean that the rare event without times set will never show up on the main page, but I think that's ok, since they are only really relevant to users once they have a start time at least.